src/main/java/org/apache/rocketmq/flink/source/reader/RocketMQPartitionSplitReader.java

```diff
@@ -120,9 +119,14 @@ public RocketMQPartitionSplitReader(
     public RecordsWithSplitIds<Tuple3<T, Long, Long>> fetch() throws IOException {
         RocketMQPartitionSplitRecords<Tuple3<T, Long, Long>> recordsBySplits =
                 new RocketMQPartitionSplitRecords<>();
-        Set<MessageQueue> messageQueues;
+        Collection<MessageQueue> messageQueues;
         try {
-            messageQueues = consumer.fetchSubscribeMessageQueues(topic);
+            messageQueues = consumer.fetchMessageQueues(topic);
+            if (StringUtils.isNotEmpty(sql)) {
+                consumer.subscribe(topic, MessageSelector.bySql(sql));
```

Comment:
Will the `fetch` method be called frequently? There may be some performance issues if so, since `subscribe` is a heavy method.

Reply:
ok, i have reoptimized the code